core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java

```diff
@@ -309,7 +309,7 @@ private boolean hasEdgeFilters(final Function<ViewElementDefinition, Boolean> ha
     }
 
     public boolean canMerge(final View addingView, final View srcView) {
-        if (addingView instanceof View && !(srcView instanceof View)) {
+        if (addingView != null && srcView == null) {
```

Comment:
Confused by this... Could you theoretically have `addingView == null && srcView != null` as you are checking the `addingView` exists?

Reply:
I was also confused by this, my commit for this change has some more notes on this. My understanding is that you could have a non-null `srcView` and a null `addingView`, these could be merged. This change does not impact the behaviour, unless I'm wrong about what `instanceof` is doing.

A merge is not possible (returning false) if there is no `srcView` but there is an `addingView`. If both are `null` then you can merge because two nulls make a null. This doesn't seem ideal to me, but as explained below, this is what the previous code did anyway.

The reason for this change is SpotBugs considers the use of `instanceof` when null checking could be done instead to be a violation. I think the original is certainly more confusing, as what is being checked is if the objects are null and not if they are an `instanceof` anything.

In this case, the only object that can be assigned to either of these is a `View` object, if it is assigned then the object is non-null, if it is not it is null. Therefore `instanceof` can be replaced with null checks which are easier to understand and probably more efficient as well.